src/karmen_frontend/src/actions/printers/reading.js

```diff
@@ -264,9 +267,35 @@ export const getWebcamSnapshot = createHttpAction(
         };
       });
     }).catch((err) => {
+      if (err instanceof StreamUnavailableError) {
+        //we got 404 from server - this means Karmen has no url to get stream from printer
+        //so we just return 404 to snapshots array so stream renderer can tell this to the user
+        //and we stop trying to get images
+        return {
+          organizationUuid: orguuid,
+          uuid,
+          status: 404,
+        };
+      }
       if (err instanceof OrganizationMismatchError) {
         return;
       }
+
+      if (err instanceof FailedToFetchDataError) {
+        //For various reasons (changing wifi, browser suspended tab on mobile, bad connection, etc), single request can fail to fetch.
+        //This causes the stream to freeze, throws generic error toaster and user has to reload the page.
+        //If we catch this case and dispatch another request after few second, we are very likely to overcome
+        //this gap and go on like nothing happen
+        //We don't have to worry about offline states - if heartbeat fails, all requests get's killed
+        setTimeout(() => dispatch(getWebcamSnapshot(orguuid, uuid)), 5000);
```

Comment:
This leaves the original timeout intact resulting in double double timers timers, wouldn't it?

Should you call `clearTimeout` first or `setWebcamRefreshInterval`?

Reply:
Original timeout is not called again if the request fails. If you comment out this line and modify backend to randomly return 502, you can open dev tools and check that after the failed request, there is 5 sec delay before another one is launched.